@here/harp-mapview/lib/poi/PoiRenderer.ts

```diff
@@ -373,37 +372,37 @@ export class PoiBatchRegistry {
     }
 }
 
-// keep track of the missing textures, but only warn once
-const missingTextureName: Map<string, boolean> = new Map();
+// keep track of the missing textures, we throw an error if the number of attempts goes over some
+// threshold.
+const missingTextureName: Map<string, number> = new Map();
+const SEARCH_CACHE_ATTEMPTS = 5;
 
 function findImageItem(
     poiInfo: PoiInfo,
     imageCaches: MapViewImageCache[],
     imageTexture?: ImageTexture
 ): ImageItem | undefined {
     assert(poiInfo.imageTextureName !== undefined);
-    const imageTextureName = poiInfo.imageTextureName!;
+    const imageTextureName = imageTexture ? imageTexture.image : poiInfo.imageTextureName!;
 
-    let imageItem: ImageItem | undefined;
     for (const imageCache of imageCaches) {
-        imageItem = imageTexture
-            ? imageCache.findImageByName(imageTexture.image)
-            : imageCache.findImageByName(imageTextureName);
+        const imageItem = imageCache.findImageByName(imageTextureName);
         if (imageItem) {
-            break;
+            missingTextureName.delete(imageTextureName);
+            return imageItem;
         }
     }
 
-    if (!imageItem) {
-        logger.error(`init: No imageItem found with name
-            '${imageTexture?.image ?? imageTextureName}'`);
+    // There is a texture missing in the cache, we attempt again, and then error out.
+    const missingTextureCount = missingTextureName.get(imageTextureName);
+    missingTextureName.set(imageTextureName, missingTextureCount ? missingTextureCount + 1 : 0);
+    if (missingTextureName.get(imageTextureName)! >= SEARCH_CACHE_ATTEMPTS) {
```

Comment:
We could log the error just once to avoid cluttering the console. Not sure if it makes sense to do it after n attempts or at the first one. It's difficult to say when "n attempts" is enough, cause it depends on how many POIs use that same image.

Reply:
I would argue that if there is an error after 5 attempts, there is something fundamentally wrong and a new approach should be taken (like the ones I mentioned in the commit message). At some point, trying to present the error in a nice way is counter productive.

Reply:
In saying that, I can reduce cluttering the console by changing the `>=` to be just `=`, then we avoid any complex state management.